internal/exec/resolvable/resolvable.go

```diff
@@ -55,8 +55,9 @@ type Field struct {
 }
 
 type FieldVisitors struct {
-	Interceptors []directives.ResolverInterceptor
-	Validators   []directives.Validator
+	ArgValidators map[string][]directives.Validator
+	Interceptors  []directives.ResolverInterceptor
+	Validators    []directives.Validator
```

Comment:
I should document these 3 use cases with appropriate example tests.
* ArgValidators is easy to understand
* Validators is for things that are not argument specific, e.g. `hasRole` (applied to fields).
* I can't see an example of resolver interceptor in this PR, unless I'm missing something. Did you mean a field directive to be applied to a (non-trivial) resolver? Can you give me an example?